### PR TITLE
chore(kubeStateMetrics): make it clear that the keys of kubeStateMetrics only accept booleans for the values

### PR DESCRIPTION
**Pull request by asdfgugus**
Discussed with @mattray. This should provide a better usability.

### CODE REVIEW

**Review by mkilchhofer — APPROVED** [review]
👍🎉

Approach inspired by my kyverno PR - i like:
https://github.com/kyverno/kyverno/pull/9532

File: charts/opencost/values.yaml [hunk 1]
```diff
@@ -259,14 +259,14 @@ opencost:
...
     kubeStateMetrics:
-      # -- Enable emission of pod annotations
-      emitPodAnnotations: ""
-      # -- Enable emission of namespace annotations
-      emitNamespaceAnnotations: ""
-      # -- Enable emission of KSM v1 metrics
-      emitKsmV1Metrics: ""
-      # -- Enable only emission of KSM v1 metrics that do not exist in KSM 2 by default
-      emitKsmV1MetricsOnly: ""
+      # -- (bool) Enable emission of pod annotations
```
Comment (mattray) [inline charts/opencost/values.yaml:262]: I'm not familiar with the `~` usage, is that doc'd somewhere (for my own curiosity)?

**Review by mattray — APPROVED** [review]

File: charts/opencost/values.yaml (hunk 1 above)
Comment (asdfgugus) [reply charts/opencost/values.yaml:262]: This is a YAML standard: https://yaml.org/spec/1.2-old/spec.html#id2805071

Comment (mkilchhofer) [reply charts/opencost/values.yaml:262]: You can also write 
`foo: null` but `foo: ~` is fancier 😆

**Comment by asdfgugus** [comment]
Can we merge?

**Pull request merged**